tests/e2e/users/add-user.specs.js

```diff
@@ -5,87 +5,80 @@ const addUserModal = require('../../page-objects/users/add-user-modal.po.js');
 
 const addedUser = 'fulltester' + new Date().getTime();
 const fullName = 'Bede Ngaruko';
-const errorMessagePassword = element(by.css('#edit-password ~ .help-block'));
 
-describe('Add user  : ', () => {
 
-  afterAll(done =>
-    utils.request(`/_users/${addedUser}`)
-      .then(doc => utils.request({
-        path: `/_users/${addedUser}?rev=${doc._rev}`,
-        method: 'DELETE'
-      }))
-      .catch(() => {}) // If this fails we don't care
-      .then(() => utils.afterEach(done)));
+describe('Add user  : ', () => {
 
-  it('should add user with valid password', () => {
-    usersPage.openAddUserModal();
-    addUserModal.fillForm(addedUser, fullName, 'StrongP@ssword1');
-    addUserModal.submit();
-    browser.wait(() => {
-      return element(by.css('#edit-user-profile')).isDisplayed()
-        .then(isDisplayed => {
-          return !isDisplayed;
-        })
-        .catch(() => {
-          return true;
-        });
-    }, 2000);
-    helper.waitForAngularComplete();
-    expect(helper.isTextDisplayed(addedUser)).toBe(true);
-    expect(helper.isTextDisplayed(fullName)).toBe(true);
+  it('should add user with valid password', async () => {
+    await usersPage.openAddUserModal();
+    await addUserModal.fillForm(addedUser, fullName, 'StrongP@ssword1');
+    await addUserModal.submit();
+    await helper.waitForTextDisplayed(addedUser);
+    await helper.waitForTextDisplayed(fullName);
+    expect(await helper.isTextDisplayed(addedUser)).toBe(true);
+    expect(await helper.isTextDisplayed(fullName)).toBe(true);
+    const userPath = `/_users/org.couchdb.user:${addedUser}`;
+    const doc = await utils.requestNative(userPath);
+    await utils.requestNative({
+      path: `${userPath}?rev=${doc._rev}`,
+      method: 'DELETE'
+    });
   });
 
-  it('should reject passwords shorter than 8 characters', () => {
-    usersPage.openAddUserModal();
-    addUserModal.fillForm('user0', 'Not Saved', 'short');
-    addUserModal.submit();
-    expect(errorMessagePassword.getText()).toBe('The password must be at least 8 characters long.');
-    element(by.css('button.cancel.close')).click();
+  it('should reject passwords shorter than 8 characters', async () => {
+    await usersPage.openAddUserModal();
+    await addUserModal.fillForm('user0', 'Not Saved', 'short');
+    await addUserModal.submit();
+    const text = await addUserModal.errorMessagePassword().getText();
+    expect(text).toBe('The password must be at least 8 characters long.');
+    await addUserModal.closeButton().click();
```

Comment:
What is this final "close" needed for? If it's because the state of the app doesn't get reset between tests, that feels kind of scary, although right now may not be the moment to try to change all that.

Otherwise, another nested `describe` block feels appropriate here, which is nice for grouping in general, but also allows for an after-action. That could do something like:
```
describe('rejections  : ', () => {
  afterEach(async () => {
    addUserModal.closeButton().click();
  });

  // all the "reject" tests go here.
});
```

Also, I recognize this is all just being migrated, so if this isn't a quick thing, please leave it for now so it doesn't block the larger progress in getting all of these running again.

Reply:
Your assumption is correct that we do no reset between tests. So the beforeAll, beforeEach, and afterAll, and afterEach are important to run.  

The `afterAll` associated to this test breaks when running individual tests because on the first actually creates a user. 

In the case you point out. I think the describe separation could be useful as a `describe('error states` or something similar. I'll add that. Should be a quick change.